### PR TITLE
Add issue templates

### PR DESCRIPTION
**Pull request by haydentherapper**
Since we've added an issue template in the repo, it means we won't inherit templates from the organization (https://github.com/sigstore/.github/tree/main/.github/ISSUE_TEMPLATE). This duplicates the bug and enhancement templates

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+name: Bug
+about: Something wrong with the project? Report it here!
+labels: 'bug'
+assignees: ''
+
+---
+
+**Description**
```
Comment (kommendorkapten) [inline .github/ISSUE_TEMPLATE/bug.md:9]: Is this the standard template for Sigstore project?

If not I would think about having sections (beyond just description and version):
* What did you observe
* What was the expected outcome
* Steps to reproduce

Comment (haydentherapper) [reply .github/ISSUE_TEMPLATE/bug.md:9]: This is the standard template from https://github.com/sigstore/.github/tree/main/.github/ISSUE_TEMPLATE, I didn't want to change it. We could update the org-wide issue templates first?

**Comment by woodruffw** [comment]
As a side-note: if we want to be even more structured here, we could use [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) instead of templates.

File: .github/ISSUE_TEMPLATE/bug.md (hunk 1 above)
Comment (haydentherapper) [reply .github/ISSUE_TEMPLATE/bug.md:9]: FWIW, the description asks for these details.

**Comment by haydentherapper** [comment]
> As a side-note: if we want to be even more structured here, we could use [issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms) instead of templates.

Oh neat! I'll take a look at that.

**Comment by woodruffw** [comment]
> Oh neat! I'll take a look at that.

This reminded me that I meant to do it for `pip-audit` months ago, so if you want some reference samples 🙂: https://github.com/pypa/pip-audit/pull/741

**Pull request closed without merging**